### PR TITLE
Add MpModelWrapper in TPU Spawn

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Add MpModelWrapper in TPU Spawn.

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by tchaton — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 1]
```diff
@@ -197,7 +201,7 @@ def broadcast(self, obj: object, src: int = 0) -> object:
         buffer = io.BytesIO()
         torch.save(obj, buffer)
         data = bytearray(buffer.getbuffer())
-        data_tensor = torch.tensor(data).to(xm.xla_device(), dtype=torch.float)
+        data_tensor = torch.tensor(data).to(self.device, dtype=torch.float)
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:202]: This avoids moving from cpu to the device

```suggestion
        data_tensor = torch.tensor(data, device=self.device, dtype=torch.float)
```

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 2]
```diff
@@ -22,6 +22,8 @@
 from torch.nn import Module
 from torch.utils.data import DataLoader
 
+import pytorch_lightning as pl
+from pytorch_lightning.overrides import LightningDistributedModule
```
Comment (awaelchli) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:26]: one of these creates cyclic import :(

**Comment by kaushikb11** [comment]
Converted to Draft, as tpu tests were failing.

File: pytorch_lightning/plugins/training_type/tpu_spawn.py [hunk 3]
```diff
@@ -257,16 +260,16 @@ def start_predicting(self, trainer) -> None:
         xmp.spawn(self.new_process, **self.xmp_spawn_kwargs)
 
     def training_step(self, *args, **kwargs):
-        return self.lightning_module.training_step(*args, **kwargs)
+        return self.model(*args, **kwargs)
```
Comment (tchaton) [inline pytorch_lightning/plugins/training_type/tpu_spawn.py:263]: This should be `wrapped_model`.

Comment (kaushikb11) [reply pytorch_lightning/plugins/training_type/tpu_spawn.py:263]: Assigning it while moving the model to device. https://github.com/PyTorchLightning/pytorch-lightning/blob/tpu_spawn_added/pytorch_lightning/plugins/training_type/tpu_spawn.py#L172

`MpModelWrapper` is just using mp lock and returns the model https://github.com/pytorch/xla/blob/master/torch_xla/distributed/xla_multiprocessing.py#L432
